addons/calendar/models/calendar_alarm.py

```diff
@@ -55,10 +56,14 @@ def _search_duration_minutes(self, operator, value):
             '&', ('interval', '=', 'days'), ('duration', operator, value / 60 / 24),
         ]
 
-    @api.onchange('duration', 'interval', 'alarm_type')
+    @api.onchange('duration', 'interval', 'alarm_type', 'notify_responsible')
     def _onchange_duration_interval(self):
         display_interval = self._interval_selection.get(self.interval, '')
         display_alarm_type = {
             key: value for key, value in self._fields['alarm_type']._description_selection(self.env)
         }[self.alarm_type]
+        if self.alarm_type not in ('sms', 'whatsapp'):
+            self.notify_responsible = False
```

Comment:
We are adding `Notify Responsible` to the name whenever `notify_responsible` is True and as `notify_responsible` has no role in Email and Notification we must make it False. Otherwise, we will get `Notify Responsible` to name whenever a user changes alarm type from SMS or WhatsApp to Email or Notification with `notify_responsible` True.